### ISSUE TITLE
simple trainning of obb object doesn't work

### ISSUE THREAD

**Issue by yuxing** [ultralytics/ultralytics#18099]
### Question

Can someone help me ? I am trainning a simple OBB example with a single object in a image, but the prediction results are very poor, the following pictures are the ground truth label and prediction

the trainning code is like this:

from ultralytics import YOLO
model = YOLO("yolo11n-obb.yaml")  # build a new model from YAML
results = model.train(data="my.yaml", epochs=100, imgsz=640)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @yuxing, thank you for bringing this up and for your interest in Ultralytics 🚀!  

We recommend checking out the [Docs](https://docs.ultralytics.com) for detailed resources and guides, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, which might help address your question more effectively.

### 🐛 Bug Reports
If this is a bug related to your setup or results, could you provide us with a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/)? This includes steps or code snippets that we can readily run to reproduce the behavior you're experiencing. Specific logs, error messages, or details about your training configuration (like `yaml` file used, command run, and `ultralytics` version) can also be very helpful!  

### ❓ Questions
If this issue is more related to custom training or model performance:  
1. Ensure you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).  
2. Providing additional details, like your dataset, the structure of your annotations, and training logs, can help us better understand your scenario.  

The Ultralytics community across [Discord](https://discord.com/invite/ultralytics) 🎧, [Discourse](https://community.ultralytics.com), or our [Subreddit](https://reddit.com/r/Ultralytics) is also a great place to share insights and get help from others working on similar tasks.

---

### Upgrade Instructions
Please ensure you're using the latest version of `ultralytics` to avoid issues already resolved in recent updates. Run the following command to upgrade:  

```bash
pip install -U ultralytics
```

The latest version automatically includes all necessary [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml). Please also confirm you are using **Python>=3.8** and **PyTorch>=1.8** as part of your environment.  

---

### Verified Environments
If the above steps don't resolve your issue, try running your experiments on one of our pre-configured environments:  
- **Notebooks with free GPU**:  
<a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a>  
<a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a>  
<a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>  

- **Google Cloud** or **Amazon AWS**:  
Refer to the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) or [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).  

- **Docker**:  
Follow the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) to set up a controlled Docker environment with the latest configurations.  
<a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>  

---

### CI Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>  

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. These tests ensure YOLO works as expected for all [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across different platforms (macOS, Windows, Ubuntu).  

---

This is an automated response 🤖, but no worries—a human-friendly Ultralytics engineer will review and assist you further shortly. Stay tuned! 😊

**Comment by Y-T-G** [comment]
You need to create a proper dataset with sufficient images for training.

https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/

**Comment by yuxing** [comment]
it seems caused by square rotatoed object detect. Does anyone know how to do with the square rotatoed object detect ?

**Comment by Y-T-G** [comment]
How many images do you have for training?

**Comment by haixiansheng** [comment]
> it seems caused by square rotatoed object detect. Does anyone know how to do with the square rotatoed object detect ?

the same problem

**Comment by glenn-jocher** [comment]
@haixiansheng for issues with detecting square rotated objects using Oriented Bounding Boxes (OBB), ensure your dataset annotations include accurate rotation angles, and that your model is trained for sufficient epochs with diverse examples. You can also experiment with adjusting hyperparameters like learning rate or augmentations. Refer to our OBB documentation for best practices: https://docs.ultralytics.com/tasks/obb.

**Comment by Y-T-G** [comment]
If you have two images, why are you starting training from scratch? And also set `nbs=1` and `batch=1` and `epochs=1000`

**Comment by yuxing** [comment]
> If you have two images, why are you starting training from scratch? And also set `nbs=1` and `batch=1` and `epochs=1000`

Ok, I will try this, thanks ~

**Comment by glenn-jocher** [comment]
You're welcome! Let us know if you encounter any issues or need further assistance while training.
